keras_nlp/samplers/sampler.py

```diff
@@ -252,20 +247,103 @@ def __call__(
 
         return tf.squeeze(prompt, axis=0) if input_is_1d else prompt
 
-    @format_docstring(sample_args=sample_args_docstring)
+    def get_next_token(self, next_token_probs):
```

Comment:
Do you think sliding window can fit into this paradigm?

Reply:
yea, it should work